config/examples/redis.yml

```diff
@@ -3,8 +3,20 @@ base: &default
   size: <%= ENV.fetch('RAILS_MAX_THREADS', 5) %>
   pool_timeout: 5 # this is in seconds
   sentinels: "<%= ENV['REDIS_SENTINEL_HOSTS'] %>"
+  sentinel_username: <%= ENV['REDIS_SENTINEL_USERNAME'].to_json %>
+  sentinel_password: <%= ENV['REDIS_SENTINEL_PASSWORD'].to_json %>
   name: <%= ENV['REDIS_SENTINEL_NAME'] %>
   role: <%= ENV['REDIS_SENTINEL_ROLE'] %>
+  # == ACL Params ==
+  username: <%= ENV['REDIS_USERNAME'].to_json %>
+  password: <%= ENV['REDIS_PASSWORD'].to_json %>
+  # == TLS Params ==
+  ssl: <%= ENV['REDIS_SSL'] %>
```

Comment:
I'm a bit confused by this setting... `redis-client` docs say:

> `ssl`: Whether to connect using SSL or not.

But I am not sure how it is actually used :thinking: 

If I set it to `false` but use `rediss://` in the URL, it still connects via SSL. So, is it really needed? :upside_down_face:  Or does it only apply when the URL is provided in a different format? e.g. `host:port` with no protocol.

Also, if this value is expected to be a boolean, I think we need to cast the env var value (which is considered a string).

I think for most of the boolean values we are using this pattern:
```
ssl: <%= ENV.fetch('REDIS_SSL', '0') == '1' %>
```

See [settings.yml](https://github.com/3scale/porta/blob/eea43d58f4e1636cb14561cfa4940de06d07c817/config/examples/settings.yml#L11)

So, it needs to be set as `REDIS_SSL=1` to be `true`.

I don't like it too much, but we have it everywhere. An alternative would be to set `REDIS_SSL=true` but then change the config to
```
ssl: <%= ActiveModel::Type::Boolean.new.cast(ENV['REDIS_SSL']) %>
```

We discussed this with @akostadinov here: https://github.com/3scale/porta/pull/3107#discussion_r1021452236

Reply:
`REDIS_SSL` is optional, because the gem is able to infer it correctly in most cases. But it doesn't infer it correctly for the sentinels + tls scenario. That scenario requires setting it manually.

I'll check the boolean format issue.

Reply:
Solved parsing `REDIS_SSL` https://github.com/3scale/porta/pull/3572/commits/3e537ba116edb95e2333a5f9f4f90bd5b632ceab